### PR TITLE
Add apparent wind angle to the route point data box

### PR DESCRIPTION
**Pull request by frye**
## Why
When inspecting a selected route point under the map, the data box showed heading, boat speed, and true wind, but it did not show apparent wind angle. That made sail trim interpretation harder during point-by-point review.

## What changed
- Derive apparent wind angle directly on `RoutePoint` from existing values (heading, boat speed, true wind speed/direction) using vector math.
- Expose both signed and absolute apparent wind angle on `RoutePoint`.
- Update `SelectedRouteDetails` to display apparent wind in `0-180°` form with side labeling (`port`/`starboard`) and explicit edge cases (`ahead`/`astern`).
- Add focused core tests for starboard, port, ahead, astern, and wraparound cases.
- Add an app workflow test asserting the selected-route data box includes apparent wind text.
- Disambiguate `Color.Parse(...)` in `MapRenderingTests` to keep app tests compiling.

## Validation
- `dotnet test tests/Navtool.Core.Tests/Navtool.Core.Tests.csproj --filter FullyQualifiedName~NativeBridgeContractTests`
- `dotnet test tests/Navtool.App.Tests/Navtool.App.Tests.csproj --filter FullyQualifiedName~MainViewModelWorkflowTests.SelectedRouteDetailsIncludeApparentWindAngle`

### CODE REVIEW

File: src/Navtool.App/ViewModels/MainViewModel.cs [hunk 1]
```diff
@@ -1273,6 +1274,23 @@ private static string FormatUtcOffset(TimeSpan offset)
...
+        }
+
+        if (roundedAngle >= 180)
+        {
+            return "apparent wind 180° astern";
+        }
+
+        var side = point.ApparentWindAngleSignedDegrees > 0d ? "starboard" : "port";
+        return $"apparent wind {roundedAngle:0}° {side}";
+    }
```
[the remark this answers is not in the document]
Comment (frye) [reply src/Navtool.App/ViewModels/MainViewModel.cs:1293]: Good catch. I changed `FormatApparentWind` to read `ApparentWindAngleSignedDegrees` once, then derive both rounded magnitude and side label from that single value. This removes duplicate trig/vector recomputation and keeps edge-case behavior consistent.


_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._
